rafx-framework/src/render_features/jobs/extract/render_feature_view_packet.rs

```diff
@@ -0,0 +1,62 @@
+use crate::render_features::render_features_prelude::*;
+use downcast_rs::{impl_downcast, Downcast};
+
+/// The index of a `RenderView` in this `RenderFeature`'s `FramePacket` for the current frame.
+pub type ViewFrameIndex = u32;
+
+/// The `ID` of a `RenderObjectInstance` in a specific `ViewPacket` for the current frame.
+pub type RenderObjectInstancePerViewId = u32;
+
+impl_downcast!(RenderFeatureViewPacket);
+/// A type-erased trait used by the `Renderer`, `RenderFrameJob`, and `RendererThreadPool`
+/// to control the workload of the rendering process without identifying specific types
+/// used in each `RenderFeature`'s frame packet or workload. See `ViewPacket` for implementation
+/// details.
+pub trait RenderFeatureViewPacket: Downcast + Sync + Send {
+    fn view(&self) -> &RenderView;
+
+    fn num_render_object_instances(&self) -> usize;
+
+    fn push_render_object_instance(
+        &mut self,
+        render_object_instance_id: RenderObjectInstanceId,
+        render_object_instance: RenderObjectInstance,
+    ) -> RenderObjectInstancePerViewId;
+
+    fn push_volume(
+        &mut self,
+        entity_id: EntityId,
+    );
+}
+
+pub trait RenderFeatureViewPacketIntoConcrete {
```

Comment:
What is this for? (comments would be good, again I'm sure you had it on your mental TODO list :) )

Reply:
It's to make this type of syntactic sugar work: 
![image](https://user-images.githubusercontent.com/4888530/118205815-12a4c300-b416-11eb-8693-cc8faa2f5e4b.png)

If I remember correctly, I had trouble getting the `into` trait to work with the generics / downcasting, so this was my workaround.

Reply:
Oh so this is essentially a downcast? Is it not possible to be a member of RenderFeatureViewPacket? If not possible, a short comment like "allows down-casting to a view packet of a specific feature" might be helpful.

Reply:
It cannot be a member of `RenderFeatureViewPacket` (or any of the other type-erased traits) because it has a generic parameter (the return value). That prevents the trait from being made into an object and then that breaks everything else. The compiler advice is to put the function into a separate trait, which is I think how we got here. It's unfortunate though. It would be significantly cleaner the other way.

Reply:
Added comments.